util/deephash/deephash.go

```diff
@@ -1010,5 +972,14 @@ func (h *hasher) hashType(t reflect.Type) {
 	// that maps reflect.Type to some arbitrary and unique index.
 	// While safer, it requires global state with memory that can never be GC'd.
 	rtypeAddr := reflect.ValueOf(t).Pointer() // address of *reflect.rtype
-	h.hashUint64(uint64(rtypeAddr))
+	h.HashUint64(uint64(rtypeAddr))
+}
+
+func unsafeStringToBytes(s string) []byte {
+	type stringHeader struct {
+		p unsafe.Pointer
+		n int
+	}
+	p := (*stringHeader)(unsafe.Pointer(&s))
+	return unsafe.Slice((*byte)(p.p), p.n)
```

Comment:
These are always super scary. This is an accidental refactor or two away from being a major problem with random crashes & corruption.

Can we instead use go4.org/mem and have sha256x have a HashMem(mem.RO) method?

Or at least make sha256x have a HashString method and do this unsafety very locally such that the []byte doesn't escape out to a caller?

Reply:
I went with adding `sha256x.Hash.HashString` instead. Now the unsafe cast is contained in just that method, which should be easier to reason about.